### PR TITLE
Thea: Fisherman Model 

### PR DESCRIPTION
**Pull request by Gauthamastro**
## Describe your changes
1. Updates Thea to Fisherman model
2. Removes the Thea aggregator dependency
3. Automatic claim of Thea deposits
4. Bug fixes to Thea authority set rotation
5. On-chain signature aggregation
6. Ethereum and other EVM chains support

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] I removed all Clippy and Formatting Warnings. 
- [x] I added required Copyrights.

### CODE REVIEW

File: pallets/thea-executor/src/lib.rs [hunk 1]
```diff
@@ -535,16 +576,13 @@ pub mod pallet {
...
-			recipient: &T::AccountId,
-		) -> Result<(), DispatchError> {
+		pub fn execute_deposit(deposit: Deposit<T::AccountId>) -> Result<(), DispatchError> {
 			// Get the metadata
 			let metadata =
 				<Metadata<T>>::get(deposit.asset_id).ok_or(Error::<T>::AssetNotRegistered)?;
 			let deposit_amount = deposit.amount_in_native_decimals(metadata); // Convert the decimals configured in metadata
 
-			if !frame_system::Pallet::<T>::account_exists(recipient) {
+			if !frame_system::Pallet::<T>::account_exists(&deposit.recipient) {
```
Comment (nuel77) [inline pallets/thea-executor/src/lib.rs:590]: If the account doesn't exist, can't we add this to a pallet account which is guaranteed to exist? this way, if any user makes the mistake of calling it without ed, the funds can still be given back to the user via governance if needed. it should be better than making the funds "vanish"

File: pallets/thea-executor/src/lib.rs [hunk 2]
```diff
@@ -520,9 +562,8 @@ pub mod pallet {
 			let deposits: Vec<Deposit<T::AccountId>> =
 				Decode::decode(&mut &payload[..]).map_err(|_| Error::<T>::FailedToDecode)?;
 			for deposit in deposits {
-				<ApprovedDeposits<T>>::mutate(&deposit.recipient, |pending_deposits| {
-					pending_deposits.push(deposit.clone())
-				});
+				// Execute Deposit
+				Self::execute_deposit(deposit.clone())?;
 				Self::deposit_event(Event::<T>::DepositApproved(
```
Comment (nuel77) [inline pallets/thea-executor/src/lib.rs:567]: ApprovedEvent is getting emitted after the claim event. This is breaking the indexer as approved event happened before claim before.

Comment (nuel77) [inline pallets/thea-executor/src/lib.rs:572]: ApprovedEvent is emitted after the claim event (within Self::execute_deposit). This is breaking indexers as the approved event happened before the claim in the previous code. If claim event is obsolete now, better to remove it.

File: pallets/thea-executor/src/lib.rs (hunk 1 above)
Comment (Gauthamastro) [reply pallets/thea-executor/src/lib.rs:590]: @zktony What do you think of sending it to pallet account ? We may be need an governance-origin extrinsic to move the funds out but that's okay.

File: pallets/thea-executor/src/lib.rs (hunk 2 above)
Comment (Gauthamastro) [reply pallets/thea-executor/src/lib.rs:572]: Claim event is there to handle pending deposits before upgrade if any, if that's not there, we can delete it now. In Fisherman model, claim process is not there anymore.

File: pallets/thea-executor/src/lib.rs (hunk 1 above)
Comment (zktony) [reply pallets/thea-executor/src/lib.rs:590]: yeah .. we can do that

Comment (Gauthamastro) [reply pallets/thea-executor/src/lib.rs:590]: Resolved.

File: pallets/thea-executor/src/lib.rs (hunk 2 above)
Comment (Gauthamastro) [reply pallets/thea-executor/src/lib.rs:572]: As discussed with @nuel77, we are going to keep it for sometime.

**Comment by serhii-temchenko** [comment]
Please let me know when this PR will be ready for review so I could update weights again after recent [changes](https://github.com/Polkadex-Substrate/Polkadex/pull/907/commits/dcd8da688f40874d2eee44e5a8ce271e1190ef4d) in benchmarks and thea-executor pallet.

**Review by serhii-temchenko — DISMISSED** [review]

**Review by serhii-temchenko — APPROVED** [review]

**Review by zktony — APPROVED** [review]

**Pull request merged**